CMakeLists.txt

```diff
@@ -24,213 +14,559 @@ set(OC_IDD_API_ENABLED ON CACHE BOOL "Enable the Introspection Device Data API."
 set(OC_TCP_ENABLED ON CACHE BOOL "Enable OCF communications over TCP. Necessary for Cloud communications.")
 set(OC_DISCOVERY_RESOURCE_OBSERVABLE_ENABLED OFF CACHE BOOL "Enable observation over oic/res resource.")
 set(OC_REPRESENTATION_REALLOC_ENCODING_ENABLED OFF CACHE BOOL "Enable realloc during encoding the representation.")
+set(OC_COLLECTIONS_IF_CREATE_ENABLED OFF CACHE BOOL "Enable RT factory for collections.")
+set(OC_MNT_ENABLED OFF CACHE BOOL "Enable maintenance resource.")
+set(OC_SOFTWARE_UPDATE_ENABLED OFF CACHE BOOL "Enable software update resource.")
+set(OC_WKCORE_ENABLED OFF CACHE BOOL "Enable well-known core resource.")
+set(OC_OSCORE_ENABLED OFF CACHE BOOL "Enable oscore support.")
+set(OC_IPV4_ENABLED OFF CACHE BOOL "Enable IPv4 support.")
+set(OC_DNS_LOOKUP_IPV6_ENABLED OFF CACHE BOOL "Enable IPv6 DNS lookup.")
+
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
[nitpick] I usually turn this only for dynamic library targets, but I don't know if there are any significant optimizations skipped in static executables/libraries because of this.

Reply:
I came across this question as well, but then I saw that the Makefile also uses PIC to compile the objects. So I switched it on globally as this matches the Makefile build.

As a future improvement we can compile the object libraries twice - once with PIC and once without.